src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,23 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> treeSet = new TreeSet<>();
+        StringBuilder stringWithUniqueCharacters = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            while (bufferedReader.ready()) {
+                int letter = bufferedReader.read();
+                if (Character.isAlphabetic(letter)) {
+                    treeSet.add((char) Character.toLowerCase(letter));
+                }
+            }
+        }
+        for (Character character : treeSet) {
+            stringWithUniqueCharacters.append(character);
+        }
+
+        return stringWithUniqueCharacters.length() < 5 ? stringWithUniqueCharacters.toString()
+                : stringWithUniqueCharacters.toString().substring(0, 5);
```

Comment:
5 is a magic number here
also, consider checking in a loop if the maximum length is exceeded